assets/js/checkFlag.js

```diff
@@ -1,12 +1,89 @@
-function checkFlag(id) {
-    let flag_input = document.getElementById(id + "-flag");
-    let flag = flag_input.value;
-    let actual_MD5 = document.getElementById(id + "-flag_MD5").innerText;
-    let flag_MD5 = hex_md5(flag);
-    
-    if(flag_MD5 == actual_MD5) {
-        flag_input.className = "form-control is-valid";
-    } else {
-        flag_input.className = "form-control is-invalid";
+(function() {
+    'use strict';
+
+    /**
+     * Storage object we use to persist solved challenge information across page loads.
+     */
+    const solvedStorage = window.localStorage;
+
+    /**
+     * Key we use to access/set solved challenges in our storage object of choice.
+     */
+    const solvedChallengesKey = "solvedChallenges";
+
+    /**
+     * Checks to see if the chall with the given name is successfully solved, according to data currently entered on the page.
+     * 
+     * We check this by comparing the MD5 hash of the given flag against the true MD5. If correct, we store that this challenge
+     * has been solved, and display this information on the page.
+     * 
+     * @param {String} challName The name of the challenge to check.
+     */
+    const checkFlag = (challName) => {
+        let challengeForm = document.getElementById(challName);
+        let flagInput = document.getElementById(`${challName}-flag`);
+        let flag = flagInput.value;
+        let correctMd5 = document.getElementById(`${challName}-flag-md5`).innerText;
+        let flagMd5 = hex_md5(flag);
+        
+        if(flagMd5.toLowerCase() === correctMd5.toLowerCase()) {
+            challengeForm.classList.remove("challenge-incorrect");
+            challengeForm.classList.add("challenge-solved");
+            storeChallengeSolved(challName);
+            return true;
+        }
+
+        challengeForm.classList.remove("challenge-solved");
```

Comment:
"Oh you solved this challenge? Well not anymore"

Reply:
I figured people may want to remove the solved status temporarily, if they want to check the flag again for whatever reason. The solve status will come back on page reload (i.e., incorrectly answering a challenge after it's been solved doesn't permanently remove the solved status).